test/integration/domain/test_result_file.py

```diff
@@ -54,7 +53,7 @@ def test_can_fetch_single_output_file(self):
         context = ExtensionContext.create("24-3144")
         result = context.output_result_file_by_id("92-5244")
         self.assertIsNotNone(result)
-        self.assertIsInstance(result, ResultFile)
+        self.assertIsInstance(result, SharedResultFile)
```

Comment:
Should this really be a `SharedResultFile`? 

Because if I trace the execution path `context.output_result_file_by_id() -> artifact_service.output_file_by_id() -> all_output_files()` I see an `assert` for a `ResultFile` on line 91 (https://github.com/Molmed/clarity-ext/blob/432d4304d6c1040b6b0e0a8f76e09d4c70895829/clarity_ext/service/artifact_service.py#L91), as well as an comparison of the type with `Artifact.OUTPUT_TYPE_RESULT_FILE`. 

There's an `all_shared_result_files()` with a comparison with the `Artifact.OUTPUT_TYPE_SHARED_RESULT_FILE` type instead, but this also has an `assert` for `ResultFile`. 

So I don't really get how this fits together.